chapters/VK_EXT_present_timing/PresentTimeInfo.txt

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2014-2020 Khronos Group.
+//
+// SPDX-License-Identifier: CC-BY-4.0
+
+[open,refpage='VkPresentTimesInfoEXT',desc='The earliest time each image should be presented',type='structs']
+--
+
+When the `<<VK_EXT_present_timing>>` extension is enabled, additional
+fields can: be specified that allow an application to specify the earliest
+time that an image should be displayed.
+This allows an application to avoid stutter that is caused by an image being
+displayed earlier than planned.
+Such stuttering can occur with both fixed and variable-refresh-rate
+displays, because stuttering occurs when the geometry is not correctly
+positioned for when the image is displayed.
+An application can: instruct the presentation engine that an image should
+not be displayed earlier than a specified time by including the
+sname:VkPresentTimesInfoEXT structure in the pname:pNext chain of the
+sname:VkPresentInfoKHR structure.
+
+The sname:VkPresentTimesInfoEXT structure is defined as:
+
+include::{generated}/api/structs/VkPresentTimesInfoEXT.txt[]
+
+  * pname:sType is the type of this structure.
+  * pname:pNext is `NULL` or a pointer to an extension-specific structure.
+  * pname:swapchainCount is the number of swapchains being presented to by
+    this command.
+  * pname:pTimes is `NULL` or a pointer to an array of
+    sname:VkPresentTimeEXT elements with pname:swapchainCount entries.
+    If not `NULL`, each element of pname:pTimes contains the earliest time
+    to present the image corresponding to the entry in the
+    sname:VkPresentInfoKHR::pname:pImageIndices array.
+
+.Valid Usage
+****
+  * [[VUID-VkPresentTimesInfoEXT-swapchainCount-01247]]
+    pname:swapchainCount must: be the same value as
+    sname:VkPresentInfoKHR::pname:swapchainCount, where
+    sname:VkPresentInfoKHR is in the pname:pNext chain of this
+    sname:VkPresentTimesInfoEXT structure.
+****
+
+include::{generated}/validity/structs/VkPresentTimesInfoEXT.txt[]
+--
+
+[open,refpage='VkPresentTimeEXT',desc='Specifying when an image should be presented',type='structs']
+--
+
+The sname:VkPresentTimeEXT union contains either a
```

Comment:
How can I distinguish these two types? C doesn't have tagged unions, so perhaps each of the target types should share a common initial field that describes which type is intended?

Reply:
I'm used to seeing the union as a struct member, where another struct member (at front of the struct) is an enum indicating which union member is the active union member. I'm not certain, but I think I've been told that accessing an inactive union member is technically undefined behavior, even if all union members are structs that begin with the same struct member (in this case an enum indicating which union member is the active one.)

Reply:
C has no concept of "active" or "inactive" union member. As far of C is concerned, any union member can be accessed and be used to re-interpret the memory contents.

Reply:
The distinction between the two types depends on the swapchain creation flags, i.e. `VK_SWAPCHAIN_CREATE_ABSOLUTE_TIME_BIT_EXT` or `VK_SWAPCHAIN_CREATE_RELATIVE_TIME_BIT_EXT`. There is no ambiguity possible from the implementation's perspective.